Source/CsDebugScript.Engine/Engine/Debuggers/DbgEngDll.cs

```diff
@@ -1299,17 +1297,34 @@ public void BreakExecution(Process process)
         /// </summary>
         public void Terminate(Process process)
         {
-            Client.EndSession((uint)Defines.DebugEndActiveTerminate);
-
             DebuggeeFlowController flowControler;
             debugeeFlowControlers.RemoveEntry(process.Id, out flowControler);
 
-            // Release any threads that are waiting.
-            //
-            flowControler.DebugStatusGo.Set();
-            flowControler.DebugStatusBreak.Set();
+            flowControler.TerminateExecution();
+        }
 
-            flowControler.WaitForDebuggerLoopToExit();
+        /// <summary>
+        /// Sets new breakpoint.
+        /// </summary>
+        /// <param name="process">Process in which breakpoint is to created.</param>
+        /// <param name="expression">Expression used to evauluate breakpoint location.</param>
+        /// <param name="executeOnBreakpointHit">Action executed when breakpoint is hit.</param>
+        /// <returns>Newly created breakpoint object.</returns>
+        public IBreakpoint SetBreakpoint(Process process, string expression, Func<BreakpointEventStatus> executeOnBreakpointHit)
```

Comment:
>SetBreakpoint [](start = 27, length = 13)

Considering that current interface is that you set breakpoint per process, why don't you save process inside the breakpoint?
Also, what happens if I debug two processes and set breakpoint to one, will other break too? Should it?